standards/naming.md

```diff
@@ -196,17 +196,17 @@ RESPONSE
 ## Domain References
 
 - Unique references to other significant entities within your API Domain, but in a different root resource **SHOULD** use a [URN (Uniform Resource Name)](https://en.wikipedia.org/wiki/Uniform_Resource_Name)-like references where appropriate. <a name="sps-ref-property-name" href="#sps-ref-property-name"><i class="fa fa-check-circle" title="#sps-ref-property-name"></i></a>
-    - URN-like refers to using the convention of specifying agreed upon namespaces, entities followed by the object identifier, but without the `urn` prefix or formal registration of the namespace ID with IANA according to [RFC8141](https://datatracker.ietf.org/doc/rfc8141/). This provides interoperability and durability benefits within your API and endpoint ecosystem.
-    - URN-like references **MUST** be the form of standard `URNs` without the `urn:` prefix to avoid confusion of official registration. A simplified example would be: `{namespace}:{entity}:{id}` (e.g. `sps:book:b96cb3ead9a9`). <a name="sps-ref-schema" href="#sps-ref-schema"><i class="fa fa-check-circle" title="#sps-ref-schema"></i></a>
+    - URN-like refers to using the convention of specifying agreed upon namespaces and entities followed by the object identifier, but without the `urn` prefix or formal registration of the namespace ID with IANA according to [RFC8141](https://datatracker.ietf.org/doc/rfc8141/). This provides interoperability and durability benefits within your API and endpoint ecosystem.
+    - URN-like references **MUST** be the form of standard `URNs` without the `urn:` prefix to avoid confusion of official registration. A simplified example would be: `{namespace}:{entity}:{sub-entity}:{id}` (e.g. `sps:book:page:b96cb3ead9a9`). <a name="sps-ref-schema" href="#sps-ref-schema"><i class="fa fa-check-circle" title="#sps-ref-schema"></i></a>
```

Comment:
You bring up a great point in the interpretation of this. Is it wrong to use it as a hierarchy? or should sub-entity only be used for classifying polymorphic objects from generic to less generic?

Reply:
It might be best to leave out the reference to RFC8141 since this change departs from the RFC significantly. See section 5 URN Namespaces:
>    Because the colon character (":") is used to separate "urn" from the
   NID and the NID from the NSS, it's tempting to think of the entire
   URN as being structured by colon characters and to assume that colons
   create a structure or hierarchy within the NSS portion of the URN.
   Such structure could be specified by a particular NID specification,
   but there is no implicit structure.  In a URN such as

>      urn:example:apple:pear:plum:cherry

>   the NSS string is `"apple:pear:plum:cherry"` as a whole, and there is
   no specific meaning to the colon characters within that NSS string
   unless such meaning is described in the specification of the
   "example" namespace.

"By the book" this scenario could be handled using the q-component (2.3.2). I'm not against the departure, but noting that departing from it reduces the value of tooling which would respect it and doesn't permit `{subentity}` to be one or none since it's just an NSS string.

Reply:
Ya our URN-like references will work in some URN parsers and not others. The main goal being for internal interoperability and context understanding. Working with the URNs in the future may require some customer parsers from SPS shared libraries. For now, they are mostly kept as opaque strings created only by the originating domain. That will need to evolve over time.

But the one key part of the comments you listed is important... that the sub-entity is actually only relevant in context with the main entity. With that understanding, sub-entity may be hierarchy or may not be... depends on how we want to use it within that entity.

I think maintaining the reference to the inspiration for SPS' URN like approach is important. While it does depart from it, much of it is a more opinionated approach, which was the desire.

Reply:
Make sense, in doing a little extra research I've found some good examples on [Wikipedia](https://en.wikipedia.org/wiki/Uniform_Resource_Name#Examples) that can lend some nice context to what the goal here is. We may actually be able to use the full URN specifier with something like a `urn:XX-sps` NID.

Reply:
While yes, everything after the second ":" is NSS - we still as an organization need to provide structure for that part that makes it standard for our internal use. It's only from overall URN spec they don't care, but we should be opinionated about the entire structure